lib/medusa/broker.ex

```diff
@@ -45,17 +50,41 @@ defmodule Medusa.Broker do
 
   def handle_cast({:publish, event, payload}, state) do
     Logger.debug "#{inspect __MODULE__}: [#{inspect event}]: #{inspect payload}"
-    Enum.each(state, fn(e) ->
-      if Regex.match?(e, event) do
-        @adapter.insert(base64_encode_regex(e), payload)
-        GenServer.cast (base64_encode_regex(e) |> String.to_atom), {:trigger}
-      end
-    end)
+    Enum.each(state, &maybe_route(&1, event, payload))
     {:noreply, state}
   end
 
 
-  def base64_encode_regex(regex) do
-    Regex.source(regex) |> :base64.encode
+  def base64_encode_name(name), do: :base64.encode name
+
+  defp maybe_route(route, incoming, payload) do
+    if route_match?(route, incoming) do
+      insert_route_to_adapter route, payload
+      trigger_producer route
+    end
+  end
+
+  defp insert_route_to_adapter(route, payload) do
+    route
+    |> base64_encode_name
+    |> @adapter.insert(payload)
```

Comment:
I think we should change this from `insert` to something more idiomatic like `enqueue`